installer/pkg/cmd/gcp_broker.go

```diff
@@ -236,17 +268,38 @@ func getOrCreateVirtualBroker(projectID, brokerName, brokerTitle string) (*virtu
 		return nil, fmt.Errorf("failed to create broker client. You might want to run 'gcloud auth application-default login'")
 	}
 
-	brokerURL := "https://servicebroker.googleapis.com"
+	host := "https://servicebroker.googleapis.com"
+	brokerURL := fmt.Sprintf("%s/v1beta1/projects/%s/brokers/%s", host, projectID, brokerName)
 	errCode, respBody, err := brokerClient.CreateBroker(&adapter.CreateBrokerParams{
-		URL:     brokerURL,
+		URL:     host,
 		Project: projectID,
 		Name:    brokerName,
 		Title:   brokerTitle,
 	})
-	if errCode == 409 {
+	if errCode == http.StatusConflict {
+		fmt.Printf("Broker %q, already exists\n", brokerName)
+		res, err := brokerClient.GetCatalog(&adapter.GetCatalogParams{
+			Server: brokerURL,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Invalid broker %q, error getting catalog for broker: %v\n", brokerURL, err)
+		}
+
+		isEAPBroker := false
```

Comment:
consider hoisting some part of the logic between 279-299 to a separate function.
Good candidate may be lines 288-299 or thereabouts.

Reply:
done.